fix issue 1053 non-string args to path.join in 0.10.x

isao opened the pull request.

Commit:
rm redundant preload() in failing test

Commit:
partial fix issue #1053; bundles without locations err on path.join()

Commit:
partial fix issue #1053; test unset path part data

also changed mock conf to always start with a clean copy

Commit:
partial fix issue #1053; pass "" to test instead of nothing

caridy commented on lib/app/autoload/store.server.js:
```diff
@@ -1692,11 +1692,11 @@ YUI.add('mojito-resource-store', function(Y, NAME) {
             switch (info.pkg.yahoo.mojito.type) {
             case 'bundle':
-                dir = this._libs.path.join(info.dir, info.pkg.yahoo.mojito.location);
+                dir = this._libs.path.join(info.dir, info.pkg.yahoo.mojito.location || '');
```
when this location member can be undefined?

isao replied:
when a user has a bundle without a location property. It's defined [here](https://github.com/yahoo/mojito/blob/develop/tests/fixtures/packages/node_modules/a/package.json) but not [here](https://github.com/yahoo/mojito/blob/develop/tests/fixtures/packages/node_modules/b/package.json) for example.

caridy replied:
So, location is not produced by the store? it just comes from the pkg information as optional then? if that's the case, we are good!

caridy commented on lib/app/autoload/store.server.js:
```diff
@@ -709,7 +709,7 @@ YUI.add('mojito-resource-store', function(Y, NAME) {
                     template = {
                         'content-path': (env === 'client' ?
-                                this._libs.path.join(this._appConfigStatic.pathToRoot || '', res.url) :
+                                this._libs.path.join(this._appConfigStatic.pathToRoot || '', res.url || '') :
```
when the res.url will be undefined/null?

isao replied:
during [this test](https://github.com/yahoo/mojito/blob/develop/tests/unit/lib/app/autoload/test-store.server.js#L500) for [this fixture](https://github.com/yahoo/mojito/tree/develop/tests/fixtures/packages).

drewfish replied:
I still think adding the real addon-rs-url in the descriptor is a cleaner fix (if it works).

caridy replied:
me too!

isao replied:
So res.url is always defined? I thought it wasn't in some cases on the client i.e. shared mojits. If no res.url is valid, then 'content-path' should not be defined with a unusable value. If the test exposes a case where res.url should be defined but isn't, then I think we should open a separate issue.

caridy replied:
Yes, we remove that exception a while ago, and we expect that url is always defined.

Commit:
partial fix issue #1053; use real addon/rs.url.js instead of mock.

back out previous `res.url || ''` guard. root cause of undefined
res.url in the unit tests was poor mock.

isao commented:
thanks for the feedback @caridy @drewfish. the root of the undefined res.url issue was test mock, not store code. unit tests pass on 0.10.2 mac.

drewfish commented:
+1
